File: 073/problem73.py
#!/usr/bin/python

"""Problem 73: Counting fractions in a range"""

import sys
sys.setrecursionlimit(10000)

def main():
    def count(n, left, right):
        med = left + right
        if med > n:
            return 0
        return 1 + count(n, left, med) + count(n, med, right)

    return count(12000, 3, 2)


if __name__ == "__main__":
    print(main())

File: 073/problem73.c
#include <stdio.h>

int main(void) {
    int limit = 12000,
        count = 0,
        top = 0,
        stack[4000],
        left = 3,
        right = 2,
        med;

    for (;;) {
        med = left + right;
        if (med > limit) {
            if (top > 0) {
                left = right;
                --top;
                right = stack[top];
            }
            else {
                break;
            }
        }
        else {
            ++count;
            stack[top] = right;
            ++top;
            right = med;
        }
    }

    printf("%d\n", count);

    return 0;
}
